src/main/java/de/tum/in/www1/artemis/web/rest/ExerciseResource.java

```diff
@@ -101,45 +100,36 @@ public ExerciseResource(ExerciseService exerciseService, ParticipationService pa
         // Exam exercise
         if (exercise.isExamExercise()) {
             Exam exam = exercise.getExerciseGroup().getExam();
-            if (authCheckService.isAtLeastEditorForExercise(exercise, user)) {
-                // instructors editors and admins should always be able to see exam exercises
-                // continue
-            }
-            else if (authCheckService.isAtLeastTeachingAssistantForExercise(exercise, user)) {
+            authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.TEACHING_ASSISTANT, exercise, user);
+            // instructors editors and admins should always be able to see exam exercises
+            if (!authCheckService.isAtLeastEditorForExercise(exercise, user)) {
```

Comment:
Can't you replace this with `authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, exercise, user);` like you did above?